final/recommendations/eclipse.tex

```diff
@@ -0,0 +1,50 @@
+The Eclipse frontend that is delivered with the product has been discussed in
+\cref{ssec:eclipse-plugin}. The plugin provides interaction with the REPL
+backend, as discussed in \cref{cha:design}.
+
+The delivered Eclipse frontend is just that: a frontend to the REPL backend.
+Integration with Eclipse and Spoofax Eclipse are not yet present. This
+subsection provides several recommendations to better integrate this frontend
+into Eclipse and Spoofax Eclipse.
+
+\subsection{Building languages and projects}
+
+To start a REPL session in the delivered project, the language designer has to
+build the language, start the REPL and issue a \texttt{``:load''} command to
+load the language. It is desirable if all this could be done automatically at
+the press of a button. Such a \textit{``Run in REPL''} button can be added to the
+``Spoofax (meta)'' submenu and inside the project- and package explorer context
+menus.
+
+When the REPL gains the ability to load existing files, such a button can also
+be added to the ``Spoofax'' submenu to assist a language user.
+
+An obstacle in implementing this feature is the classpath issue as described in
+\cref{sec:classpath}.
+
+\subsection{Back and forth interaction between Eclipse and the REPL}
+
+A feature the REPL does not yet support is loading existing files into the
+evalution context (see \cref{ssec:discuss-repl}). When it does, this interaction
+could provide deeper integration between the REPL and Spoofax Eclipse. For
+example, hovering over variables could indicate from where they originate, or
+changes in a currently loaded file could be automatically picked up. Changes in
+the other direction are interesting, too: when a loaded function is overridden
+inside the REPL, an option could be provided to apply this change to the loaded
+file as well.
+
+\subsection{Improving the UI of the Eclipse REPL}
+
+The user interface currenty offered by the Eclipse frontend does not resemble a
+typical REPL: the input and output views are separated. To improve the user
+experience, these two views could be merged into one. This would require the use
+of a widget or text representation that can be \textit{``semi-read-only''}, as
+the previously entered text and results should not be editable, while the
+current input should be.
+
+TODO: recommend missed features such as autocompletion, hovers, et cetera?
```

Comment:
I'd just leave this TODO out tbh. You've already shortly mentioned hovers somewhere else.